src/vote.rs

```diff
@@ -44,6 +44,11 @@ impl<T: Serialize + Clone> Vote<T> {
         })
     }
 
+    /// Compose a Vote.
+    pub(crate) fn compose(payload: T, signature: Signature) -> Self {
```

Comment:
We've spoken about making a module for the data chain stuff before.  If we did that now, we could reduce the visibility of this method to just that module.

Reply:
That would be even better.